Server error <mixtral>:  [CRITICAL] WORKER TIMEOUT  ....    [ERROR] Worker (pid:#) was sent SIGKILL! Perhaps out of memory?

I saw the below while attempting a run against mixtral using PR #107 

```
[2024-03-25 21:53:27 -0400] [11801] [CRITICAL] WORKER TIMEOUT (pid:12903)
[2024-03-25 21:53:28 -0400] [11801] [ERROR] Worker (pid:12903) was sent SIGKILL! Perhaps out of memory?
[2024-03-25 21:53:28 -0400] [13179] [INFO] Booting worker with pid: 13179
```

Server console output:  https://gist.github.com/jwmatthews/32cc1dedd8ecfb2c4364e38533b43fe7

--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ run-postgres:
 
 # Can adjust logging level by defining env var: KAI_LOG_LEVEL
 run-server:
-	PYTHONPATH=$(KAI_PYTHON_PATH) gunicorn --timeout 1200 -w $(NUM_WORKERS) --bind localhost:8080 --worker-class aiohttp.GunicornWebWorker 'kai.server:app("$(LOGLEVEL)")'
+	PYTHONPATH=$(KAI_PYTHON_PATH) gunicorn --timeout 3600 -w $(NUM_WORKERS) --bind localhost:8080 --worker-class aiohttp.GunicornWebWorker 'kai.server:app("$(LOGLEVEL)")'
 
 load-data:
 	PYTHONPATH=$(KAI_PYTHON_PATH) python ./kai/incident_store_advanced.py  --config_filepath ./kai/config.toml

--- a/example/run_demo.py
+++ b/example/run_demo.py
@@ -84,7 +84,7 @@ def _generate_fix(params: KaiRequestParams):
         f"{SERVER_URL}/get_incident_solutions_for_file",
         data=params.to_json(),
         headers=headers,
-        timeout=2400,
+        timeout=3600,
     )
     return response
 

--- a/kai/server.py
+++ b/kai/server.py
@@ -30,6 +30,9 @@ from kai.prompt_builder import PromptBuilder
 from kai.pydantic_models import parse_file_solution_content
 from kai.report import Report
 
+LLM_RETRIES = 5
+LLM_RETRY_DELAY = 10
+
 # TODO: Make openapi spec for everything
 
 # TODO: Repo lives both on client and on server. Determine either A) Best way to
@@ -430,7 +433,7 @@ async def get_incident_solutions_for_file(request: Request):
             f"Processing incident {count}/{len(request_json['incidents'])} for {incident['file_name']}"
         )
         llm_output = None
-        for _ in range(3):
+        for _ in range(LLM_RETRIES):
             try:
                 llm_output = get_incident_solution(request.app, incident, False)
                 content = parse_file_solution_content(llm_output.content)
@@ -439,9 +442,9 @@ async def get_incident_solutions_for_file(request: Request):
                 break
             except Exception as e:
                 KAI_LOG.warn(
-                    f"Request to model failed with exception, retrying in 10s\n{e}"
+                    f"Request to model failed for {incident['file_name']} {count}/{len(request_json['incidents'])} with exception, retrying in {LLM_RETRY_DELAY}s\n{e}"
                 )
-                time.sleep(10)
+                time.sleep(LLM_RETRY_DELAY)
 
     end = time.time()
     KAI_LOG.info(

--- a/kai/server.py
+++ b/kai/server.py
@@ -429,11 +429,19 @@ async def get_incident_solutions_for_file(request: Request):
         KAI_LOG.info(
             f"Processing incident {count}/{len(request_json['incidents'])} for {incident['file_name']}"
         )
-        llm_output = get_incident_solution(request.app, incident, False)
-        content = parse_file_solution_content(llm_output.content)
-
-        total_reasoning.append(content.reasoning)
-        current_file_contents = content.updated_file
+        llm_output = None
+        for _ in range(3):
+            try:
+                llm_output = get_incident_solution(request.app, incident, False)
+                content = parse_file_solution_content(llm_output.content)
+                total_reasoning.append(content.reasoning)
+                current_file_contents = content.updated_file
+                break
+            except Exception as e:
+                KAI_LOG.warn(
+                    f"Request to model failed with exception, retrying in 10s\n{e}"
+                )
+                time.sleep(10)
 
     end = time.time()
     KAI_LOG.info(

